collector/nginx.go

```diff
@@ -14,14 +16,16 @@ type NginxCollector struct {
 	logger      *slog.Logger
 	nginxClient *client.NginxClient
 	metrics     map[string]*prometheus.Desc
+	timeout     time.Duration
 	mutex       sync.Mutex
 }
 
 // NewNginxCollector creates an NginxCollector.
-func NewNginxCollector(nginxClient *client.NginxClient, namespace string, constLabels map[string]string, logger *slog.Logger) *NginxCollector {
+func NewNginxCollector(nginxClient *client.NginxClient, namespace string, constLabels map[string]string, logger *slog.Logger, timeout time.Duration) *NginxCollector {
```

Comment:
Could this function provide a sensible default for the `timeout`? Currently it breaks public API.